static/app/utils/profiling/hooks/useProfileEvents.tsx

```diff
@@ -11,13 +11,14 @@ import RequestError from 'sentry/utils/requestError/requestError';
 import useApi from 'sentry/utils/useApi';
 import useOrganization from 'sentry/utils/useOrganization';
 import usePageFilters from 'sentry/utils/usePageFilters';
+import {ProfilingFieldType} from 'sentry/views/profiling/profileSummary/content';
 
 type Sort<F> = {
   key: F;
   order: 'asc' | 'desc';
 };
 
-interface UseProfileEventsOptions<F> {
+export interface UseProfileEventsOptions<F = ProfilingFieldType> {
```

Comment:
Should we be using extends here actually?

Reply:
yeah, i would say both... i don't always need to pass in the generic type so the assignment makes it optional, but yes we should also extend so that we ensure the field types is one of the pre-defined vals